src/harmony/bin/run_harmony.R

```diff
@@ -5,54 +5,52 @@ print("# Harmony: Algorithm for single cell integration #")
 print("##################################################")
 
 # Loading dependencies scripts
+library("argparse")
 
-library("optparse")
-parser <- OptionParser(
-  prog = "run_harmony.R",
-  description = "Scalable integration of single cell RNAseq data for batch correction and meta analysis"
+parser <- ArgumentParser(description='Scalable integration of single cell RNAseq data for batch correction and meta analysis')
+parser$add_argument(
+    'input',
+    metavar='INPUT',
+    type="character",
+    help='Input file [default]'
 )
-parser <- add_option(
-  parser,
-  c("-i", "--input-file"),
-  action = "store",
-  default = NULL,
-  help = "Input file [default]"
+parser$add_argument(
+    '--output-prefix',
+    type="character",
+    dest='output_prefix',
+    default = "foo",
+  	help="Prefix path to save output files. [default %default]"
 )
-parser <- add_option(
-  parser,
-  c("-a", "--vars-use"),
-  action = "store",
-  default = NULL,
-  help = "If meta_data is dataframe, this defined which variable(s) to remove (character vector)."
+parser$add_argument(
+    "--vars-use",
+    type="character",
+    dest='vars_use',
+    action="append",
+	  default=NULL,
+    help='If meta_data is dataframe, this defined which variable(s) to remove (character vector).'
 )
-parser <- add_option(
-  parser,
-  c("-p", "--do-pca"),
-  action = "store",
-  default = FALSE,
-  help = "Whether to perform PCA on input matrix."
+parser$add_argument(
+    '--seed',
+    type="character",
+    dest='seed',
+    default=617,
+    help='Seed. [default %default]'
 )
-parser <- add_option(
-  parser,
-  c("-o", "--output-prefix"),
-  action = "store",
-  default = "foo",
-  help="Prefix path to save output files. [default %default]"
-)
-parser <- add_option(
-  parser, 
-  c("-s", "--seed"), 
-  action = "store", 
-  default = 617,
-  help="Seed. [default %default]"
+parser$add_argument(
+	'--do-pca',
+	type="logical",
+	dest='do_pca',
+	action="store",
+	default=FALSE,
+	help='Whether to perform PCA on input matrix.'
 )
 
-args <- parse_args(parser)
+args <- parser$parse_args()
 
 cat("Parameters: \n")
 print(args)
 
-if(is.null(args$`vars-use`)) {
+if(is.null(args$`vars_use`)) {
```

Comment:
It would be nice to remove all of the backticks around the variable names now that you've switched to underscores.

Reply:
Done (Indeed these are not needed)